app/controllers/application_controller/filter.rb

```diff
@@ -1793,4 +1784,15 @@ def process_datetime_selector(param_key_suffix, exp_key = nil)
     exp[exp_value_key][exp_value_index] = "#{date}#{time}"
   end
   private :process_datetime_selector
+
+  ENABLE_CALENDAR = "miqBuildCalendar();".freeze
+  def calendar_needed?
+    return true if [:date, :datetime].include?(@edit.fetch_path(@expkey, :val1, :type))
+    return true if [:date, :datetime].include?(@edit.fetch_path(@expkey, :val2, :type))
+    false
+  end
+  included do
+    helper_method :calendar_needed?  # because it is being called inside a render block
+  end
```

Comment:
It will be good idea to move it on top of the file, so if needed other methods can be added later to that block.